psm_lib.py

```
@@ -33,36 +37,143 @@ def calculate_propensity_score(df, treatment_col, covariate_cols):
+    
+    fig = go.Figure()
+
+    # Add reference lines
+    fig.add_vline(x=0.1, line_dash="dash", line_color="gray", annotation_text="0.1 (Target)")
+    fig.add_vline(x=0, line_color="black")
+
+    # Add Unmatched points (Red Circles)
+    fig.add_trace(go.Scatter(
+        x=df_plot['SMD_Unmatched'], y=df_plot['Variable'],
+        mode='markers', name='Unmatched',
+        marker=dict(color='red', size=10, symbol='circle')
```

Ruff (0.14.10)

Unnecessary `dict()` call (rewrite as a literal)

Rewrite as a literal

(C408)